packages/dm-core-plugins/src/form/templates/ObjectStorageUncontainedTemplate.tsx

```diff
@@ -28,59 +36,56 @@ export const ObjectStorageUncontainedTemplate = (props: TObjectTemplate) => {
-  const canOpen =
-    referenceExists &&
-    onOpen &&
-    (uiAttribute?.functionality?.open ?? config.functionality.open)
+
+  const { canExpand, canOpenInTab } = InferCanOpenOrExpand(
+    referenceExists,
+    config,
+    uiAttribute
+  )
 
   return (
-    <Fieldset>
-      <Legend>
-        <Typography bold={true}>{getDisplayLabel(attribute)}</Typography>
-        {!config.readOnly && (
-          <SelectReference
-            attributeType={attribute.attributeType}
-            namePath={namePath}
-          />
-        )}
-        {attribute.optional && address && !config.readOnly && (
-          <RemoveObject address={address} namePath={namePath} />
-        )}
-        {canOpen && (
-          <OpenObjectButton
-            viewId={namePath}
-            viewConfig={
-              uiAttribute?.openViewConfig
-                ? uiAttribute?.openViewConfig
-                : {
-                    type: 'ReferenceViewConfig',
-                    scope: namePath,
-                    recipe: uiAttribute?.uiRecipe,
-                  }
-            }
-            idReference={address}
-          />
-        )}
-      </Legend>
-      {canExpand && (
-        <ViewCreator
-          idReference={address}
-          onOpen={onOpen}
-          viewConfig={
-            uiAttribute?.expandViewConfig
-              ? uiAttribute?.expandViewConfig
-              : {
-                  type: 'ReferenceViewConfig',
-                  recipe: uiAttribute?.uiRecipe,
-                }
+    <FormObjectBorder>
+      <ObjectLegendWrapper>
+        <ObjectLegendHeader
+          canExpand={canExpand}
+          canOpenInTab={canOpenInTab}
+          isExpanded={isExpanded}
+          attribute={attribute}
+          objectIsNotEmpty={referenceExists}
+          setIsExpanded={setIsExpanded}
+          openInTab={() =>
+            onOpen?.(namePath, OpenViewConfig(uiAttribute), address)
           }
-          onChange={(data: any) => setValue(namePath, data)}
         />
+        <ObjectLegendActionsWrapper>
+          {canOpenInTab && referenceExists && (
+            <OpenObjectButton
+              viewId={namePath}
+              viewConfig={OpenViewConfig(uiAttribute)}
+              idReference={address}
+            />
+          )}
+          {!config.readOnly && (
+            <SelectReference
+              attributeType={attribute.attributeType}
+              namePath={namePath}
+            />
+          )}
+          {attribute.optional && referenceExists && !config.readOnly && (
+            <RemoveObject address={address} namePath={namePath} />
+          )}
+        </ObjectLegendActionsWrapper>
+      </ObjectLegendWrapper>
+      {canExpand && referenceExists && isExpanded && (
+        <FormExpandedViewWrapper>
+          <ViewCreator
+            idReference={address}
+            onOpen={onOpen}
+            viewConfig={ExpandViewConfig(uiAttribute)}
+            onChange={(data: any) => setValue(namePath, data)}
```

Comment:
Could you properly type this?

Reply:
I'm actually not sure how to do that, because this field can be any object. 
What do you suggest?

Reply:
Right, didn't think about that being the view creator... I guess `Record<string,unknown>` would be more appropriate, though, since its an object with unknown content :+1: